### ISSUE TITLE
Home screen on gallery needs improved a11y navigation

### ISSUE THREAD

**Issue by dnfield** [material-components/material-components-flutter-gallery#134]
## Acceptance Criteria
* Swipe traversal works in a sensible way
* The "crane" section is reachable via a11y navigation
* Expanding buttons/menus does not disrupt other lists/make announcements about being in a new list of more items than actually available.

## Reproduction steps

Run the gallery on a physical Android device with Talkback enabled.  Start swiping left/right to navigate around. Note:

- [ ] Cards for demos don't scroll when activated
- [ ] Cannot reach the Crane item at all
- [x] Expanding any of the buttons along the bottom announces in a list of 6 items
- [x] Swiping beyond those items leads to an invisible list (I think it's the menu list?)

/cc @willlarche @goderbauer @rami-a @darrenaustin @vick08

**Comment by willlarche** [comment]
@dnfield is this still oustanding?

**Comment by dnfield** [comment]
I believe the carousel still needs the new options that weren't available when this was opened. We could file a new bug for just that if it helps.

**Comment by dnfield** [comment]
We needed changes from https://github.com/flutter/flutter/pull/45598 which won't be available until next stable.
